Userland/Utilities/uniq.cpp

```diff
@@ -5,75 +5,106 @@
  */
 
 #include <AK/RefPtr.h>
+#include <AK/StringView.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 #include <unistd.h>
 
-struct linebuf {
-    char* buf = NULL;
-    size_t len = 0;
-};
-
-static FILE* get_stream(char const* filepath, char const* perms)
+static ErrorOr<void> write_line_content(StringView line, size_t count, bool duplicates_only, bool print_count, Core::Stream::File& outfile)
 {
-    FILE* ret;
+    if (duplicates_only && count <= 1)
+        return {};
 
-    if (filepath == nullptr) {
-        if (perms[0] == 'r')
-            return stdin;
-        return stdout;
-    }
+    if (print_count)
+        TRY(outfile.write(String::formatted("{} {}\n", count, line).bytes()));
+    else
+        TRY(outfile.write(String::formatted("{}\n", line).bytes()));
+    return {};
+}
 
-    ret = fopen(filepath, perms);
-    if (ret == nullptr) {
-        perror("fopen");
-        exit(1);
+static StringView skip(StringView line, unsigned char_skip_count, unsigned field_skip_count)
+{
+    line = line.trim("\n"sv);
+    if (field_skip_count) {
+        bool in_field = false;
+        int field_index = 0;
+        unsigned current_field = 0;
+        for (size_t i = 0; i < line.length(); i++) {
+            char c = line[i];
+            if (c == ' ' || c == '\t' || c == '\v' || c == '\f' || c == '\r') {
```

Comment:
isn't this just `is_ascii_whitespace` from `AK/CharacterType.h` ?

Reply:
didn't know that was a thing, fixed

Reply:
(it's is_ascii_space btw)